src/status_im2/contexts/chat/events.cljs

```diff
@@ -294,20 +297,70 @@
 
 (rf/defn mute-chat-toggled-successfully
   {:events [:chat/mute-successfully]}
-  [{:keys [db]} chat-id muted-till]
+  [{:keys [db]} chat-id muted-till mute-type muted? chat-type]
   (log/debug "muted chat successfully" chat-id " for" muted-till)
-  {:db (assoc-in db [:chats chat-id :muted-till] muted-till)})
+  (let [time-string (fn [duration-kw unmute-time]
+                      (str (i18n/label duration-kw)
+                           \newline
+                           "("
+                           (i18n/label :until)
+                           unmute-time
+                           ")"))
+        not-community-chat? (chat-utils/not-community-chat? chat-type)
+        mute-duration-text
+        (fn [unmute-time]
+          (if unmute-time
+            (str
+             (case mute-type
+               constants/mute-for-15-mins-type (time-string
```

Comment:
@ibrkhalil, recently @cammellos faced a painful problem to debug during development because of the usage of `case` with vars (code that requires evaluation).

I learned my lesson too. We should really NOT use vars in the clauses because they're not evaluated. See the docstring, this is first sentence:

> The test-constants are not evaluated.

For example, you could type `banana` as one of the clauses, and the compiler would not catch the undefined var, even though outside the macro CLJS fails to compile code with undefined vars. It's a slippery slope we should avoid.

The following code returns `:default-value` and no warning or error is generated in CLJS or CLJ. This is not a bug in the language, it's just that we should not use `case` like this.

```clojure
(let [mute-type :xyz]
  (case mute-type
    constants/mute-until-the-universe-ends "this will never match or warn"
    constants/mute-for-15-mins-type "this will never match or warn"
    :default-value))
```

The solution is to use `condp` in this example, and leave `case` only for its intended purpose, i.e. to match literal values. So let's eliminate all usages of `case` like this from the code if we see them.

cc'ing some folks to share the lesson @flexsurfer, @OmarBasem, @J-Son89, @ulisesmac

Reply:
Thanks for the wonderfully detailed explanation, I Used `condp`.